spec/models/provider_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe Provider, type: :model do
+
+  describe "Provider" do
+    it "is provider valid" do
+      expect(build(:provider)).to be_valid
+    end
+  end
+
+  describe "Validations" do
+    it { is_expected.to validate_uniqueness_of(:name) }
+  end
+
+  describe "Associations" do
+    it "has many warehouse_items" do
+      assc = described_class.reflect_on_association(:warehouse_items)
+      expect(assc.macro).to eq :has_many
+    end
+
+    it "has many warehouse_item_providers" do
+      assc = described_class.reflect_on_association(:warehouse_item_providers)
+      expect(assc.macro).to eq :has_many
+    end
+  end
+
+  describe "before_save" do
+     it { is_expected.to callback(:set_name).before(:save) }
+  end
+
+  describe "priority" do
+   it "enumerator value" do
+     expect(build(:provider, priority: 0).priority).to eq("contract")
+   end
+
+   it "enumerator value" do
+     expect(build(:provider, priority: 1).priority).to eq("individual_sale")
+   end
+    it { is_expected.to define_enum_for(:priority) }
+  end
+
+  describe "active" do
+    it "active true" do
```

Comment:
Recuerda que la descripcion debe "describir" lo que esperas. En este caso esperas que el scope "active" regrese los registros con active = true. Entonces el describe seria "active scope" en lugar de solo "active" y el it seria "returns reccords where 'active' = true" ó "returns reccords where 'active' is true"